java-backend/src/main/java/org/kframework/backend/java/symbolic/SymbolicRewriter.java

```diff
@@ -883,6 +902,21 @@ private KItem getCell(KItem root, String label) {
         return null;
     }
 
+    public ConstrainedTerm buildHaltTerm(ConstrainedTerm term) {
+        KItem result;
+        try {
+            List<Term> cells = global.javaExecutionOptions.haltCells.stream()
+                    .map(name -> getCell((KItem) term.term(), String.format("<%s>", name)))
+                    .collect(Collectors.toList());
+            result = KItem
+                    .of(KLabelConstant.of(KLabels.HALT_CELLS_CELL, global.getDefinition()), KList.concatenate(cells),
+                            global);
```

Comment:
Is this extra `HALT_CELLS_CELL` cell added as a "dummy" cell to append the other cells to? I don't understand the purpose.

Reply:
Yes, exactly.

Reply:
Let's try to do this another way. I guess if a user has a cell named `<haltCells>` (however improbable), this will cause problems, and special-cased cell names now will only be a problem for us in the future.